be/src/vec/aggregate_functions/aggregate_function_group_array_intersect.h

```diff
@@ -0,0 +1,526 @@
+template <typename T>
+class AggregateFunctionGroupArrayIntersect
+        : public IAggregateFunctionDataHelper<AggregateFunctionGroupArrayIntersectData<T>,
+                                              AggregateFunctionGroupArrayIntersect<T>> {
+private:
+    using State = AggregateFunctionGroupArrayIntersectData<T>;
+    DataTypePtr argument_type;
+
+public:
+    AggregateFunctionGroupArrayIntersect(const DataTypes& argument_types_)
+            : IAggregateFunctionDataHelper<AggregateFunctionGroupArrayIntersectData<T>,
+                                           AggregateFunctionGroupArrayIntersect<T>>(
+                      argument_types_),
+              argument_type(argument_types_[0]) {}
+
+    AggregateFunctionGroupArrayIntersect(const DataTypes& argument_types_,
+                                         const bool result_is_nullable)
+            : IAggregateFunctionDataHelper<AggregateFunctionGroupArrayIntersectData<T>,
+                                           AggregateFunctionGroupArrayIntersect<T>>(
+                      argument_types_),
+              argument_type(argument_types_[0]) {}
+
+    String get_name() const override { return "group_array_intersect"; }
+
+    DataTypePtr get_return_type() const override { return argument_type; }
+
+    bool allocates_memory_in_arena() const override { return false; }
+
+    void add(AggregateDataPtr __restrict place, const IColumn** columns, ssize_t row_num,
+             Arena*) const override {
+        auto& data = this->data(place);
+        auto& init = data.init;
+        auto& set = data.value;
+
+        const bool col_is_nullable = (*columns[0]).is_nullable();
+        const ColumnArray& column =
+                col_is_nullable ? assert_cast<const ColumnArray&>(
+                                          assert_cast<const ColumnNullable&>(*columns[0])
+                                                  .get_nested_column())
+                                : assert_cast<const ColumnArray&>(*columns[0]);
+
+        const auto data_column = column.get_data_ptr();
+        const auto& offsets = column.get_offsets();
+        const size_t offset = offsets[static_cast<ssize_t>(row_num) - 1];
+        const auto arr_size = offsets[row_num] - offset;
+
+        using ColVecType = ColumnVector<T>;
+        const auto& column_data = column.get_data();
+
+        const bool is_column_data_nullable = column_data.is_nullable();
+        const ColumnNullable* col_null = nullptr;
+        const ColVecType* nested_column_data = nullptr;
+
+        if (is_column_data_nullable) {
+            auto const_col_data = const_cast<IColumn*>(&column_data);
+            col_null = static_cast<ColumnNullable*>(const_col_data);
+            nested_column_data = &assert_cast<const ColVecType&>(col_null->get_nested_column());
+        } else {
+            nested_column_data = &static_cast<const ColVecType&>(column_data);
+        }
+
```

Comment:
This method is too complicated. Maybe some logic can be move into the member function of AggregateFunctionGroupArrayIntersectData?

Reply:
done